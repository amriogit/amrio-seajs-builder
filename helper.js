var OP = Object.prototype
var AP = Array.prototype

var helper = {
    type: function(obj) {
        return OP.toString.call(obj).slice(8, -1)
    },
    isObject: function(obj) {
        return helper.type(obj) === 'Object'
    },
    isArray: function(array) {
        return Array.isArray(array)
    },
    extend: function() {
        var args = AP.slice.call(arguments, 0)
        var dest = args.shift()
        args.forEach(function(src) {
            if (!src) return
            Object.keys(src).map(function(k) {
                if (helper.isObject(dest[k]) && helper.isObject(src[k])) {
                    helper.extend(dest[k], src[k])

                } else if (Array.isArray(dest[k]) && Array.isArray(src[k])) {
                    dest[k].concat(src[k])
                    
                } else {
                    dest[k] = src[k]
                }
            })
        })
        return dest
    },
    color: function(text) {
        return util.inspect(text, {
            colors: true
        })
    },
    unique: function(list) {
        var result = []
        list.forEach(function(item) {
            result.indexOf(item) === -1 && result.push(item)
        })
        return result
    },
    asyncEach: function(list, iterator, done) {
        var index = 0

        function next() {
            if (index >= list.length) {
                done()
            } else {
                iterator(list[index++], next)
            }
        }

        next()
    }
}

module.exports = helper
