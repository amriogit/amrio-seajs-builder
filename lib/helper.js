var util = require('util')

var OP = Object.prototype
var AP = Array.prototype

var helper = {
    noop: function() {},
    type: function(obj) {
        return OP.toString.call(obj).slice(8, -1)
    },
    isObject: function(obj) {
        return helper.type(obj) === 'Object'
    },
    isArray: function(array) {
        return Array.isArray(array)
    },
    extend: function() {
        var args = AP.slice.call(arguments, 0)
        var dest = args.shift()

        if (!dest) {
            return dest
        }

        args.forEach(function(src) {
            if (!src) return
            Object.keys(src).map(function(k) {
                if (helper.isObject(dest[k]) && helper.isObject(src[k])) {
                    helper.extend(dest[k], src[k])

                } else if (Array.isArray(dest[k]) && Array.isArray(src[k])) {
                    dest[k] = dest[k].concat(src[k])

                } else {
                    dest[k] = src[k]
                }
            })
        })
        return dest
    },
    color: function(text) {
        return util.inspect(text, {
            colors: true
        })
    },
    unique: function(list) {
        var result = []
        list.forEach(function(item) {
            result.indexOf(item) === -1 && result.push(item)
        })
        return result
    },
    asyncEachSeries: function(array, iterator, callback) {
        var index = 0
        callback && (callback = helper.noop)

        function next() {
            if (index >= array.length) {
                callback()
            } else {
                var isCalled = false
                iterator(array[index++], function() {
                    isCalled || next()
                    isCalled = true
                })
            }
        }

        next()
    },
    asyncEach: function(array, iterator, callback) {
        var index = 0
        var len = array.length
        callback && (callback = helper.noop)

        function done() {
            if (++index >= len) {
                callback()
            }
        }

        if (array.length === 0) {
            callback()
        } else {
            array.forEach(function(item) {
                var isCalled = false
                iterator(item, function() {
                    isCalled || done()
                    isCalled = true
                })
            })
        }
    }
}

module.exports = helper
