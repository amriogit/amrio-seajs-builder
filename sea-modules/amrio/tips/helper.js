define("amrio/tips/helper",[],function(e,i,r){"use strict";r.exports={helper:{}}});
