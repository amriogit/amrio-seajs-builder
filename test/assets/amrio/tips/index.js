define(function(require, exports, module) {
    'use strict'
    // require('./style.css')
    require('./helper')
    
    var tools = require('amrio/tools/index')

    module.exports = {
        show: function(text) {
            text && tools.log(text)
        }
    }
})