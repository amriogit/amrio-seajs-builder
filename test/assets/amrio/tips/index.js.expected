define("amrio/tips/index", [ "$", "bootstrap", "amrio/tools/index" ], function(require, exports, module) {
    "use strict";
    var $ = require("$");
    var bootstrap = require("bootstrap");
    require("./style.css");
    require("./helper");
    var tools = require("amrio/tools/index");
    module.exports = {
        show: function(text) {
            text && tools.log(text);
        }
    };
});

define("amrio/tips/style.css", [], function() {
<<<<<<< HEAD
    seajs.importStyle('@import url("./base.css");\n\n.ui-tips{\n    font-size: 12px;\n    border: 1px #f00 solid;\n}\n.ui-tips-content{\n    background-color: #fff;\n}');
=======
    seajs.importStyle('@import url("./base.css");\r\n\r\n.ui-tips{\r\n    font-size: 12px;\r\n    border: 1px #f00 solid;\r\n}\r\n.ui-tips-content{\r\n    background-color: #fff;\r\n}');
>>>>>>> 0b618e8fc9a45856c0629bac77cf2f65dcb8e964
});

define("amrio/tips/helper", [], function(require, exports, module) {
    "use strict";
    module.exports = {
        helper: {}
    };
});